src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterService.cs

```diff
@@ -100,6 +109,37 @@ public class ImporterService : IImporterService
             GeographicLevel.PlanningArea
         };
 
+        private static readonly List<GeographicLevel> AllowedSoloGeographicLevels = new List<GeographicLevel>
+        {
+            GeographicLevel.Provider,
+            GeographicLevel.School,
+        };
+
+        public static bool HasSoloAllowedGeographicLevel(HashSet<GeographicLevel> subjectGeographicLevels)
+        {
+            if (subjectGeographicLevels.Count == 1)
+            {
+                foreach (var allowedSoloLevel in AllowedSoloGeographicLevels)
+                {
+                    if (subjectGeographicLevels.Contains(allowedSoloLevel))
+                    {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+        public static bool AllowRowImport(HashSet<GeographicLevel> subjectGeographicLevels, GeographicLevel rowGeographicLevel)
+        {
+            if (HasSoloAllowedGeographicLevel(subjectGeographicLevels))
```

Comment:
I think there's a potential optimisation that can be made here. Not sure if it would have a big impact on import time but it might be worth doing.

So to begin with, this is the same as

```
            return HasSoloAllowedGeographicLevel(subjectGeographicLevels) ||
                   !IgnoredGeographicLevels.Contains(rowGeographicLevel);
```

The result of `HasSoloAllowedGeographicLevel(subjectGeographicLevels)` never changes yet this `AllowRowImport` method is called for every csv row within loops in `CopyRows`, `ImportFiltersAndLocations`, `GetObservations`, so potentially millions of tiems. In each of those places, before entering the loop you could calculate the result of  `HasSoloAllowedGeographicLevel(subjectGeographicLevels)` and if true, you don't need to call `AllowRowImport` at all.